src/main/java/cat/udl/eps/softarch/unicloud/domain/Subject.java

```diff
@@ -0,0 +1,38 @@
+package cat.udl.eps.softarch.unicloud.domain;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import java.math.BigDecimal;
+
+
+@Entity
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class Subject extends UriEntity<Long>{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```

Comment:
You have defined the identifier from the extended UriEntity as `Long` but then the field id is implemented of type `long` (lower case) which is different. Use `Long` in both places.